pkg/sql/plan/shuffle.go

```diff
@@ -35,6 +37,37 @@ const (
 	ShuffleToMultiMatchedReg int32 = 2
 )
 
+// convert first 8 bytes to uint64
+func ByteSliceToUint64(bytes []byte) uint64 {
+	var result uint64 = 0
+	i := 0
+	length := len(bytes)
+	for ; i < 8; i++ {
+		result = result * 256
+		if i < length {
+			result += uint64(bytes[i])
+		}
+	}
+	return result
+}
+
+// convert first 8 bytes to uint64. vec.area must be nil
+// if varlena length less than 8 bytes, should have filled zero in varlena
+func VarlenaToUint64Inline(v *types.Varlena) uint64 {
```

Comment:
change the parameter type to *byte, thus we can use it for both varlena and "area"